### PR TITLE
Make export templates extraction handle subdirectories

### LINKED ISSUE #25330: Mono export templates extraction issue
**Godot version:**
3.1 Beta 2 Mono

**OS/device including version:**
Manjaro Linux x86_64

**Issue description:**
When downloading export templates they are extracted incorrectly resulting in all contents in the topmost folder for the specified template.
image

### PR DESCRIPTION
**Pull request by neikeq**
Fixes #25330

### CODE REVIEW

File: editor/export_template_manager.cpp [hunk 1]
```diff
@@ -283,6 +285,24 @@ bool ExportTemplateManager::_install_from_file(const String &p_file, bool p_use_
...
+			base_dir = base_dir.substr(templates_dir.length(), file_path.length()).trim_prefix("/").trim_prefix("\\");
+			file = base_dir.plus_file(file);
+
+			DirAccessRef da = DirAccess::create(DirAccess::ACCESS_FILESYSTEM);
+			ERR_CONTINUE(!da);
+
+			String output_dir = template_path.plus_file(base_dir);
+
+			if (!DirAccess::exists(output_dir)) {
+				Error err = da->make_dir_recursive(output_dir);
```
Comment (akien-mga) [inline editor/export_template_manager.cpp]:
```
editor/export_template_manager.cpp:301:11: error: declaration of 'err' shadows a previous local [-Werror=shadow=local]
```

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Comment by bruvzg** [comment]
- Open Windows startup folder to witness the fun - `Start`->`Run`->`shell:startup`
- Download and install attached "export templates" (`.tpz` inside `.zip` cause GitHub doesn't like files with unknown extensions)
- ...
- Profit!

[completely_legit_export_templates_for_windows.tpz.zip](https://github.com/godotengine/godot/files/2904958/completely_legit_export_templates_for_windows.tpz.zip)

Obviously using export templates from unknown sources is not the right thing to do, but maybe we should sanitize filenames a bit before extracting (at least remove nasty stuff like `..`).

**Comment by akien-mga** [comment]
It was the same before, though, no? Would be worth opening a separate bug report.

**Comment by bruvzg** [comment]
> It was the same before, though, no?

No previously it was extracting everything into the same folder without taking care of the path.

**Pull request merged**